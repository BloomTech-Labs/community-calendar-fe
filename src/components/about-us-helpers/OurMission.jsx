import React from 'react'

import {
  section1,
  section2,
  section3,
  section4,
  div1,
  div2,
  div3,
} from './OurMission.module.scss'

const OurMission = () => {
  return (
    <>
      <section id={section1}>
        <div>
          <h2 className='is-bold is-size-3 is-family-secondary'>
            OUR
            <br />
            MISSION
          </h2>
        </div>
        <div>
          <p data-testid='mission-statement'>
            Our platform should make discovering a local event on any given day
            as easy and as rewarding as finding something new to watch on
            Netflix. A local event app serving municipalities and neighborhoods.
            See the events happening in your local area in an intuitive way.
          </p>
        </div>
      </section>

      <section id={section2}>
        <div>
          <div />
        </div>
      </section>

      <section id={section3}>
        <div className={div1} />
        <div className={div2}>
          <h3
            className='is-family-secondary is-size-3'
            data-testid='target-audience'
          >
            Target Audience
          </h3>
          <p>
            We are targeting real people in communities
            <br />
            who find it hard to find local events.
          </p>
        </div>
        <div className={div3} />
      </section>

      <section id={section4}>
        <div className={div1}>
          <h3 className='is-family-secondary is-size-3' data-testid='our-focus'>
            Our Focus
          </h3>
          <p>
            To make easier to find local events in their community and
            respective location.
            <br />
            We aim to solve that by creating a design
            <br />
            that is elegant and intuitive.
          </p>
        </div>
        <div className={div2} />
      </section>
    </>
  )
}

export default OurMission
