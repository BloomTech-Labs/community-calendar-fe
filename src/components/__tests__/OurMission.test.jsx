import React from 'react'
import OurMission from '../about-us-helpers/OurMission.jsx'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom'

describe('Tests for OurMission.jsx', () => {
  describe(`Snapshot`, () => {
    test('should match snapshot', () => {
      const tree = render(<OurMission />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('test if component renders our properly', () => {
    test('should display our focus, mission, and target audience correctly', () => {
      const {getByTestId} = render(<OurMission />)
      const focusSection = getByTestId('our-focus')
      const targetAudienceSection = getByTestId('target-audience')
      const missionStatement = getByTestId('mission-statement')

      expect((focusSection, targetAudienceSection, missionStatement))
        .toBeInTheDocument
    })
  })
})
