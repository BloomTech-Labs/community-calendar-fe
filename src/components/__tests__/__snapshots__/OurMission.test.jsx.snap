// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for OurMission.jsx Snapshot should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="section1"
      >
        <div>
          <h2
            class="is-bold is-size-3 is-family-secondary"
          >
            OUR
            <br />
            MISSION
          </h2>
        </div>
        <div>
          <p
            data-testid="mission-statement"
          >
            Our platform should make discovering a local event on any given day as easy and as rewarding as finding something new to watch on Netflix. A local event app serving municipalities and neighborhoods. See the events happening in your local area in an intuitive way.
          </p>
        </div>
      </section>
      <section
        id="section2"
      >
        <div>
          <div />
        </div>
      </section>
      <section
        id="section3"
      >
        <div
          class="div1"
        />
        <div
          class="div2"
        >
          <h3
            class="is-family-secondary is-size-3"
            data-testid="target-audience"
          >
            Target Audience
          </h3>
          <p>
            We are targeting real people in communities
            <br />
            who find it hard to find local events.
          </p>
        </div>
        <div
          class="div3"
        />
      </section>
      <section
        id="section4"
      >
        <div
          class="div1"
        >
          <h3
            class="is-family-secondary is-size-3"
            data-testid="our-focus"
          >
            Our Focus
          </h3>
          <p>
            To make easier to find local events in their community and respective location.
            <br />
            We aim to solve that by creating a design
            <br />
            that is elegant and intuitive.
          </p>
        </div>
        <div
          class="div2"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="section1"
    >
      <div>
        <h2
          class="is-bold is-size-3 is-family-secondary"
        >
          OUR
          <br />
          MISSION
        </h2>
      </div>
      <div>
        <p
          data-testid="mission-statement"
        >
          Our platform should make discovering a local event on any given day as easy and as rewarding as finding something new to watch on Netflix. A local event app serving municipalities and neighborhoods. See the events happening in your local area in an intuitive way.
        </p>
      </div>
    </section>
    <section
      id="section2"
    >
      <div>
        <div />
      </div>
    </section>
    <section
      id="section3"
    >
      <div
        class="div1"
      />
      <div
        class="div2"
      >
        <h3
          class="is-family-secondary is-size-3"
          data-testid="target-audience"
        >
          Target Audience
        </h3>
        <p>
          We are targeting real people in communities
          <br />
          who find it hard to find local events.
        </p>
      </div>
      <div
        class="div3"
      />
    </section>
    <section
      id="section4"
    >
      <div
        class="div1"
      >
        <h3
          class="is-family-secondary is-size-3"
          data-testid="our-focus"
        >
          Our Focus
        </h3>
        <p>
          To make easier to find local events in their community and respective location.
          <br />
          We aim to solve that by creating a design
          <br />
          that is elegant and intuitive.
        </p>
      </div>
      <div
        class="div2"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
